tests/components/temper/test_sensor.py

```diff
@@ -3,15 +3,21 @@
 from datetime import timedelta
 from unittest.mock import Mock, patch
 
+from freezegun.api import FrozenDateTimeFactory
+
 from homeassistant.core import HomeAssistant
 from homeassistant.setup import async_setup_component
 from homeassistant.util import dt as dt_util
+from homeassistant.util.unit_system import METRIC_SYSTEM
 
 from tests.common import async_fire_time_changed
 
 
-async def test_temperature_readback(hass: HomeAssistant) -> None:
+async def test_temperature_readback(
+    hass: HomeAssistant, freezer: FrozenDateTimeFactory
+) -> None:
     """Test for reading sensors."""
+    hass.config.units = METRIC_SYSTEM
```

Comment:
@jbouwh the problem is not with the time freeze, but with other tests overriding the default system, your PR will still fail if you run the following:
`pytest tests/components/lg_thinq/test_climate.py tests/components/temper/test_sensor.py`

background in https://github.com/home-assistant/core/pull/140295

This specific test actually doesn't have anything wrong, there are other tests that modify the `METRIC_SYSTEM` constant

I will tag you on discord since we try to think of the optimal way to prevent this in the future.

Reply:
hmm